runelite-client/src/main/java/net/runelite/client/ui/overlay/Overlay.java

```diff
@@ -27,21 +27,26 @@
 import java.awt.Dimension;
 import java.awt.Point;
 import java.awt.Rectangle;
+import java.util.ArrayList;
+import java.util.List;
 import lombok.Getter;
 import lombok.Setter;
 import net.runelite.client.ui.overlay.components.LayoutableRenderableEntity;
+import net.runelite.client.ui.overlay.menu.MenuOption;
 
 @Getter
 @Setter
 public abstract class Overlay implements LayoutableRenderableEntity
 {
+	private String overlayName = null;
```

Comment:
You should make this overridable method probably, or use the other method that already exists

Reply:
This holds the Menu Target name to display, you can `setOverlayName()` inside an overlay to update this value. If this is null it defaults to the existing `getName()`. 

I didn't want to update the original because It is used for config storage.

Reply:
Storing keys with spaces is fine, so it should be fine and at least there is some default value instead of null.